lib/api.js

```diff
@@ -515,6 +521,62 @@ api.declare({
   res.reply(prices);
 });
 
+api.declare({
+  method: 'delete',
+  route: '/credentials',
+  name: 'claimCredentials',
+  title: 'Request worker credentials',
+  stability: API.stability.experimental,
+  input: 'claim-credentials-request.json#',
+  output: 'claim-credentials-response.json#',
+  description: [
+    'Yield credentials for the worker running in the given instance.',
+  ].join(' '),
+}, async function(req, res) {
+  const sIdentDoc = new Buffer(req.body.identityDocument, 'base64').toString();
+  const signature = new Buffer(req.body.signature, 'base64').toString();
+
+  const E = () => {
+    return res.reportError('AuthorizationFailed', 'Failure to verify authorization');
+  };
+
+  if (!await validateIdentityDocument(sIdentDoc, signature)) {
+    return E();
+  }
+
+  const identDoc = JSON.parse(sIdentDoc);
+  const instances = await this.state.listInstances({
+    id: identDoc.instanceId,
+    region: identDoc.region,
+  });
+
+  if (instances.length != 1) {
+    return E();
+  }
+
+  const instance = instances[0];
+  if (instance.state != 'running' || instance.hasClaimedCredentials) {
+    return E();
+  }
+
+  const provisionerId = this.keyPrefix.slice(0, this.keyPrefix.length - 1);
```

Comment:
I think this patch is most of the work toward removing the need for a keyPrefix? Is it time to add a `provisionerId` config?

Reply:
OK, and at that point we'd have a more straightforward way to get a provisionerId?  This just seems like an odd way to find that value.